### DISCUSSION TITLE
Welcome to miniPaint Discussions!

### DISCUSSION THREAD

**Discussion by viliusle** [viliusle/miniPaint#216] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Mekaboo** [comment]
Hello!
I'm  writing because I wanted to see if it was possible to change the look  of MiniPaint to fit my website. Basically all I want is the middle area  and the drawing tools which will be turned into a bottom navbar instead  of a left sidebar. A pic image sidebar gallery  to be used to draw on  will be where the  left drawing tools are now.  Is it possible? Wanted  to ask before I downloaded and tried to implement. I can be reached at coti032@gmail.com. Thank ya

**Comment by viliusle** [comment]
Hello,
I do not adapt or customize miniPaint for specific use case. I only implement what is useful for all users. 
You will need to try to do it yourself, check https://github.com/viliusle/miniPaint/wiki/Build-instructions

p.s. we have themes.

**Comment by D0nVitalio** [comment]
Hello. Could you please tell how to edit images stored on server and save them back to server?

**Reply by viliusle** [reply]
You have to use AJAX to pass data from server to client, then modify data, and then send it back to server using AJAX.
There are some examples about miniPaint usage:
https://github.com/viliusle/miniPaint/tree/master/examples

**Comment by zhaohanfeng** [comment]
Hi Viliusle,

miniPaint is a great tool. I would like to use it in my project for users to change photo background. Is it OK for me to modify remove the miniPaint logo and hide some items of the menu when I embed miniPaint in my web page?

I really appreciate your great work and I would like to make donation to this project. Is there any way to do so? Thank you.

**Reply by viliusle** [reply]
Hello, it is ok.

**Reply by zhaohanfeng** [reply]
Many thanks.

**Comment by frijolitox** [comment]
Hi man, i test the examples in miniPaint-photo_editor\examples, my problem is, the browser give error "Uncaught DOMException: Permission denied to access property "Layers" on cross-origin object"

![error_mini_paint](https://user-images.githubusercontent.com/12910292/168975708-88b4fc19-c367-4968-9fa9-8acfae1bcccb.png)
Can you help me please
Thank you

**Reply by viliusle** [reply]
Hello,
You are probably trying to use miniPaint using iframe, and iframe target is https://viliusle.github.io/miniPaint/, you should download sources and set iframe target to be in same server, where you are working.

**Reply by frijolitox** [reply]
Hi Man,
Thank's for your response.
My problem is that I can't use minipaint outside your code, can you help me please, my code is attached
[test.zip](https://github.com/viliusle/miniPaint/files/8737120/test.zip)
.

**Reply by frijolitox** [reply]
ps: i have your source code, i download it and i use with my test.tml

**Reply by frijolitox** [reply]
![error_mini_paint](https://user-images.githubusercontent.com/12910292/169513821-a3f167fc-9a82-453d-9b34-0d513922e312.png)

**Reply by viliusle** [reply]
This is wrong place, you have issue, and there is place for [issues](https://github.com/viliusle/miniPaint/issues). This is place for simple questions, or discussions.

p.s. I just downloaded miniPaint source, copied to my local server, and everything works fine. I do not want to add more details here because this is really wrong place to do it.

**Comment by SPARQLCRMSUPPE** [comment]
Hi! Thanks for this fabulous tool!!

**Comment by maxgag51** [comment]
Hi Vilius, first of all, congratulations for this work.
I have some questions.
1. Do you plan to implement free selection functions in addition to the rectangular selection in the future?
2. What exactly are the key-points used for? I thought they could be useful to create a path, to measure the number of points or the length of the paths but I don't see results in this sense
3. Do you plan to extend the key-points so that they also work within a selected area, without having to create a new level?
4. Please note that when you create a new level from a cropped area, the key-points also take into account the cropping rectangle and not just the points in the underlying image.

**Reply by viliusle** [reply]
Hi,

1. I do not plan to do it soon, it would require too much development time.
2. This is experimental tool. Key points can be be used for image/face recognition and comparison of multiple images. Basically it marks most important points in current image.
3. No, because it is old and experimental tool, not even sure if anybody else are using it.
4. Yeah, I see it. Sadly, because of things mentioned in third part, priority for this issue is very low.

**Reply by viliusle** [reply]
4 issue is fixed on master branch

**Reply by maxgag51** [reply]
Great!

**Comment by ghost** [comment]
Hi everyone.

**Reply by viliusle** [reply]
Hi

**Comment by xuan-cn** [comment]
I run server'ing
error:
Uncaught Error: Module build failed (from ./node_modules/babel-loader/lib/index.js):
TypeError: D:\Desktop\2023\20230316\miniPaint-master\src\js\core\gui\gui-tools.js: Property left of ForInStatement expected node to be of a type ["VariableDeclaration","LVal"] but instead got undefined

**Reply by xuan-cn** [reply]
what error.....not runing..

**Reply by viliusle** [reply]
Hello, looks like you have some issue, and this is place for discussions. I will not give you any more help here only because it is wrong place. Please use [issues](https://github.com/viliusle/miniPaint/issues).
p.s. I just tried to build app from zero on Windows and worked. Means it needs more details / testing.

**Comment by toduckhai** [comment]
Hello
I am very impressed with your project
However, I still haven't figured out how you build it
Did you use any frameworks in this project and which tools did you use the most?
In your project I see there is an image search function, did you download these photos or were they downloaded directly from the internet?

**Reply by viliusle** [reply]
Hello
>Did you use any frameworks in this project and which tools did you use the most?

I did not use any frameworks, but tried to build clear structure to keep things clear so even after adding much more features, code is still maintainable. Result is not bad, even there are few small issues.

>In your project I see there is an image search function, did you download these photos or were they downloaded directly from the internet?

I am using https://pixabay.com/ service, they are being downloaded directly from their API, when users enter search term and submits form.
